packages/extension-dapp/src/index.ts

```diff
@@ -176,3 +176,22 @@ export async function web3FromAddress (address: string): Promise<InjectedExtensi
 
   return web3FromSource(found.meta.source);
 }
+
+// retrieve a provider from one source
+export async function web3RpcProvider (source: string, providerJSON: ProviderJSON): Promise<InjectedProviderWithMeta | null> {
```

Comment:
Just source here, no JSON or setting. (It should not be driven by the user, rather the extension to set all info)